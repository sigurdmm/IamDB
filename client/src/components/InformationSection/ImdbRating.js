import PropTypes from 'prop-types';
import React from 'react';

const ImdbRating = ({ rating }) => <h3>{`Imdb rating: ${rating}`}</h3>;

ImdbRating.propTypes = {
  rating: PropTypes.string,
};

ImdbRating.defaultProps = {
  rating: 'Unknown',
};

export default ImdbRating;
