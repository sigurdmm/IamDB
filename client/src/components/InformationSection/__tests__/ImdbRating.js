import React from 'react';
import { render } from 'enzyme';
import ImdbRating from '../ImdbRating';

const mockedRating = {
  existingRating: 8.8,
  noRating: {},
};

describe('<ImdbRating/>', () => {
  it('should render rating when rating is set', () => {
    const wrapper = render(<ImdbRating rating={mockedRating.existingRating}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
