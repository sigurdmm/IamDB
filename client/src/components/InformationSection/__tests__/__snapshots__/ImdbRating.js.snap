// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImdbRating/> should render rating when rating is set 1`] = `
<h3>
  ⭐️: 8.8
</h3>
`;
